FirebaseStorage/Sources/Internal/StorageGetDownloadURLTask.swift

```diff
@@ -74,10 +74,10 @@ class StorageGetDownloadURLTask: StorageTask, StorageTaskManagement {
              .jsonObject(with: data) as? [String: Any] {
             downloadURL = self.downloadURLFromMetadataDictionary(responseDictionary)
             if downloadURL == nil {
-              self.error = NSError(domain: StorageErrorDomain,
-                                   code: StorageErrorCode.unknown.rawValue,
-                                   userInfo: [NSLocalizedDescriptionKey:
-                                     "Failed to retrieve a download URL."])
+              self.error = StorageError.unknown(
+                message: "Failed to retrieve a download URL.",
+                serverError: [:]
+              ) as NSError
```

Comment:
Nvm, it's because of the property type of `self.error`. It might be able to get switched to `StorageError` which should bridge to `NSError` given the `CustomNSError` conformance, but I don't think it's necessary.

Reply:
Just to confirm because some other code paths like 69 and 83 throw `NSError`s, they can be caught as `StorageError`s?

Reply:
I tried to change, but the variable needs to be compatible with the NSError's coming from GTMSessionFetcher

Reply:
@paulb777, does that affect the ability to catch it as a `StorageError` in Swift?

Reply:
Verified that the ` return StorageError.unknown(message: "File at URL: \(fileURL?.absoluteString ?? "") is " +
      "not reachable. Ensure file URL is not " +
      "a directory, symbolic link, or invalid url.",
      serverError: [:]) as NSError` in StorageUploadTask's contentUploadError gets caught in the `testAttemptToUploadDirectoryShouldFail` test as `catch let StorageError.unknown(reason, _)`